mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp

```diff
@@ -525,20 +557,53 @@ class MatMulConverter : public OpConversionPattern<tosa::MatMulOp> {
       dynDims[2] = rewriter.create<tensor::DimOp>(loc, op->getOperand(1), 2);
     }
 
+    auto getTypeWithoutBatch = [&](ShapedType ty) {
```

Comment:
Will this work when dims are dynamic?

Reply:
It does work, as long as `getReassociationIndicesForReshape` can resolve the indices. I have tests for that.